java/Nikiforov-Oleksandr/src/com/zoolatech/lecture3/tasks/_4/CsvReporter.java

```diff
@@ -0,0 +1,20 @@
+package com.zoolatech.lecture3.tasks._4;
+
+public class CsvReporter extends Reporter {
+
+    public CsvReporter(String headerData, String fileData, String fileName) {
+        super(headerData, fileData, fileName);
+    }
+
+    @Override
+    void generate() {
+        openFile();
+        System.out.println(".txt");
```

Comment:
`fileName` should already contain a file extension, output statements before opening and closing a file can be removed.